libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -363,8 +472,10 @@ static int init_model_ov(OVModel *ov_model, const char *input_name, const char *
             goto err;
         }
     }
-
-    status = ie_core_load_network(ov_model->core, ov_model->network, ctx->options.device_type, &config, &ov_model->exe_network);
+    if (exec_params->in_frame->format == AV_PIX_FMT_VAAPI)
+        status = init_model_ov_vaapi(ov_model, exec_params->in_frame, input_name);
+    else
+        status = ie_core_load_network(ov_model->core, ov_model->network, ctx->options.device_type, &config, &ov_model->exe_network);
```

Comment:
These if else clause looks confusing.
I get that the load_model with vaapi input would need extra code, but `init_model_ov_vaapi` is not a good function name, since it looks like` init_model_ov` but another path.  Maybe load_network_vaapi or something like it is better?

Reply:
Right, `load_network_vaapi` would be better. I will modify it.